utils_nlp/models/transformers/named_entity_recognition.py

```diff
@@ -1,2 +1,430 @@
 # Copyright (c) Microsoft Corporation. All rights reserved.
-# Licensed under the MIT License.
\ No newline at end of file
+# Licensed under the MIT License.
+
+import torch
+import logging
+import numpy as np
+
+from torch.utils.data import TensorDataset
+from cached_property import cached_property
+from collections import Iterable
+
+from transformers.modeling_bert import (
+    BERT_PRETRAINED_MODEL_ARCHIVE_MAP,
+    BertForTokenClassification
+)
+
+from utils_nlp.models.transformers.common import (
+    MAX_SEQ_LEN,
+    TOKENIZER_CLASS,
+    Transformer,
+    get_device,
+)
+
+TC_MODEL_CLASS = {k: BertForTokenClassification for k in BERT_PRETRAINED_MODEL_ARCHIVE_MAP}
+
+
+class TokenClassificationProcessor():
+    """
+    Process raw dataset for training and testing, and post-process prediction
+    for token classification.
+
+    Args:
+        model_name (str, optional): The pretained model name.
+            Defaults to "bert-base-cased".
+        to_lower (bool, optional): Lower case text input.
+            Defaults to False.
+        cache_dir (str, optional): The default folder for saving cache files.
+            Defaults to ".".
+    """
+
+    def __init__(self, model_name="bert-base-cased", to_lower=False, cache_dir="."):
+        self.model_name = model_name
+        self.to_lower = to_lower
+        self.cache_dir = cache_dir
+        self.tokenizer = TOKENIZER_CLASS[model_name].from_pretrained(
+            model_name, do_lower_case=to_lower, cache_dir=cache_dir
+        )
+
+    @staticmethod
+    def get_inputs(batch, model_name, train_mode=True):
+        """
+        Produce a dictional object for model training or prediction.
+
+        Args:
+            model_name (str): The pretained model name.
+            train_mode (bool, optional): Whether it's for model training. Set it to False if
+                it's for testing and it won't have the 'labels' data field.
+                Defaults to True, for model training.
+        
+        Returns:
+            A dictional object contains all needed information for training or testing.
```

Comment:
Typo in dictionary. :) I believe the exact docstring format for returned values is:
dict:  A dictionary object contains...

Reply:
Updated. Thanks.